framework/include/meshgenerators/RefineSidesetGenerator.h

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MeshGenerator.h"
+
+/**
+ * MeshGenerator for refining one or more sidesets
+ */
+class RefineSidesetGenerator : public MeshGenerator
+{
+public:
+  static InputParameters validParams();
+
+  RefineSidesetGenerator(const InputParameters & parameters);
+
+protected:
+  virtual std::unique_ptr<MeshBase> generate() override;
+
+private:
+  /// Input mesh to refine
+  std::unique_ptr<MeshBase> & _input;
+
+  /// List of boundarie(s) to refine
+  const std::vector<BoundaryName> _boundaries;
+
+  /// The amount of times to refine each boundary, corresponding to their index in 'boundaries'
+  const std::vector<int> _refinement;
+
+  /// Toggles whether neighboring level one elements should be refined or not. Defaults to true.
+  const bool _enable_neighbor_refinement;
+
+  /// Side(s) of the boundary/boundaries to be refined. Can be either "primary"(just the boundary elements), "secondary"(just the neighboring elements), or "both."
+  const MultiMooseEnum _boundary_side;
+
+  /// The actual function refining the boundaries. This is done recursively in order to minimize the number of refinement iterations.
+  virtual std::unique_ptr<MeshBase>
```

Comment:
I don't believe this needs to be virtual

Reply:
This is just copied from earlier generate functions which were all virtual, if it doesn't affect anything negatively to make it virtual we might as well just leave it.